### PR TITLE
feat(authentication): implement change email

### PR DESCRIPTION
**Pull request by ChrisPdgn**
**What kind of change does this PR introduce?** 

- [ ] Bugfix
- [x] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other (please describe)

**Does this PR introduce a breaking change?** 

- [ ] Yes
- [x] No

**Other information:**
Verification is now done through hooks, after bugfix logically an email will be sent to user

### CODE REVIEW

File: modules/authentication/src/handlers/local.ts [hunk 1]
```diff
@@ -122,6 +122,22 @@ export class LocalHandlers implements IAuthenticationStrategy {
       this.changePassword.bind(this),
     );
 
+    routingManager.route(
+      {
+        path: '/local/change-email',
+        action: ConduitRouteActions.POST,
+        description: `Changes the user's email but requires password first.
+                      If 2FA is enabled then a message will be returned asking for token input.`,
```
Comment (kon14) [inline modules/authentication/src/handlers/local.ts:130]: Remove 2FA reference from description.

**Review by kon14 — CHANGES_REQUESTED** [review]

**Comment by kon14** [comment]
This is going to have to be rebased against #257 (or vice-versa).

**Review by kon14 — APPROVED** [review]

File: modules/authentication/src/handlers/local.ts [hunk 2]
```diff
@@ -545,6 +572,89 @@ export class LocalHandlers implements IAuthenticationStrategy {
...
+          email: newEmail,
+        },
+      });
+      if (this.sendVerificationEmail) {
+        const serverConfig = await this.grpcSdk.config.get('router');
+        const url = serverConfig.hostUrl;
+        const result = { verificationToken, hostUrl: url };
+        const link = `${result.hostUrl}/hook/authentication/verify-change-email/${result.verificationToken.token}`;
+        await this.emailModule
+          .sendEmail('EmailVerification', {
```
Comment (kkopanidis) [inline modules/authentication/src/handlers/local.ts:640]: Use the new template name you created, not the old one

File: modules/authentication/src/handlers/local.ts [hunk 3]
```diff
@@ -545,6 +572,89 @@ export class LocalHandlers implements IAuthenticationStrategy {
...
+      );
+    }
+    const passwordsMatch = await AuthUtils.checkPassword(
+      password,
+      dbUser.hashedPassword!,
+    );
+    if (!passwordsMatch) {
+      throw new GrpcError(status.UNAUTHENTICATED, 'Invalid password');
+    }
+    const dupEmailUser = await User.getInstance().findOne({ email: newEmail });
```
Comment (kkopanidis) [inline modules/authentication/src/handlers/local.ts:613]: make this check after the check of line 590, before checking password validation. In  general you should try and keep a parameters validation in the same place and not in various locations throughout the code.

**Review by kkopanidis — CHANGES_REQUESTED** [review]

**Review by kkopanidis — APPROVED** [review]

**Pull request merged**